libraries/mongodb.rb

```diff
@@ -39,7 +48,14 @@ def self.configure_replicaset(node, name, members)
     begin
       connection = nil
       rescue_connection_failure do
-        connection = Mongo::Connection.new('localhost', node['mongodb']['config']['port'], :op_timeout => 5, :slave_ok => true)
+        connection = self.get_connection(node)
+        if node['mongodb']['config']['auth'] == true
+          begin
+            connection.add_auth("admin", node['mongodb']['admin']['username'], node['mongodb']['admin']['password'])
+          rescue Mongo::AuthenticationError => e
+            Chef::Log.warn("Unable to authenticate as admin user. If this is a fresh install, ignore warning: #{e}")
```

Comment:
First and foremost, I'm trying to guess why this warning exists at all, instead of a plain (fatal) error.
From the warn message I assume in some cases (for example fresh install) we authentication to fail. Why ?

I then assumed this could happen if the cookbook user asks the cookbook to create the admin user, in addition to create the replicaset, and we may not know in which order both operations take place, thus the uncertainty about the admin authentication working.
It may well not be the scenario you had in mind when writing this warn log line, or even not a possible scenario (I'm not familiar with the user creation part of this cookbook).